developer_docs/smart_proxy_dev_setup.asciidoc

```diff
@@ -0,0 +1,149 @@
+= Smart Proxy devel setup
+:toc: right
+:toclevels: 5
+
+[[prerequisites]]
+== Prerequisites
+https://github.com/theforeman/smart-proxy[Smart Proxy] will run with the following requirements (aside from rubygem dependencies):
+
+* Ruby `2.7.x`
+* PostgreSQL 13
+
+[[installation]]
+== Installation
+
+[source, bash]
+....
+git clone https://github.com/theforeman/smart-proxy
+cd smart-proxy
+cp config/settings.yml.example config/settings.yml
+bundle install
+....
+
+[[configuration]]
+== Configuration
+
+By default, Smart Proxy does not listen on any port and SSL is not configured, so uncomment the `http_port` configuration value:
+
+[source, bash]
+....
+# config/settings.yaml
+:http_port: 8080
+:foreman_url: http://localhost:3000
+:bind_host: ['*']
+....
+
+To elevate authorization for Foreman, either the remote IP address or DNS hostname, or both, must be in the trusted_hosts list. This depends on how your system connects to the proxy (IPv4 or IPv6) and also on how the remote IP address is reverse-resolved. The following list of trusted entries should do the trick:
+
+[source, bash]
+....
+grep -A10 trusted_hosts config/settings.yaml
+:trusted_hosts:
+- localhost
+- localhost4
+- localhost6
+- 127.0.0.1
+- ::1
+- your_host_name
+- your_host_name.with.full.domain.com
+....
+
+[[running-proxy]]
+== Running proxy
+[source, bash]
+....
+bundle exec bin/smart-proxy
+....
+
+[[smart-proxy-in-foreman]]
+== Smart Proxy in Foreman
+Add proxy in Foreman UI (_Infrastructure > Smart Proxies_)
+
+[source, bash]
+....
+Name: localhost
+URL: http://localhost:8000
```

Comment:
Is it 8000 or 8080? (as in the config/settings.yaml file)

Reply:
The default for is 8000 when http is used